### PR TITLE
Conflicting policies

### PR DESCRIPTION
**Pull request by theri**
Added text about what happens when an application specifies both protocol and path selection properties. Path selection properties have precedence. I added examples to both API and implementation draft to illustrate this. Relates to #149

### CODE REVIEW

File: draft-brunstrom-taps-impl.md [hunk 1]
```diff
@@ -328,6 +328,9 @@ Branching for derived endpoints is the final step, and may have multiple layers
 Implementations should sort the branches of the tree of connection options in order of their preference rank. 
 Leaf nodes on branches with higher rankings represent connection attempts that will be raced first.
 Implementations should order the branches to reflect the preferences expressed by the application for its new connection, including Protocol and Path Selection Properties, which are specified in {{I-D.trammell-taps-interface}}. 
+Path Selection Properties take precedence over Protocol Selection Properties.
+
+If Protocol and Path Selection Properties contain any prohibited properties, the implementation should purge branches containing nodes with these properties.
```
Comment (britram) [inline draft-brunstrom-taps-impl.md]: Does this mean that Prohibit takes precedence over all others? (I think it does, and I think it's correct). I think what I'd like to see here (and maybe I'm just a languages geek) is a strict ordering, e.g. something like:

`Prohibit` > `Require` > `Prefer` > `Avoid`

File: draft-trammell-taps-interface.md [hunk 2]
```diff
@@ -361,6 +361,13 @@ transport protocol selection is necessarily tied to path selection. This may
 involve choosing between multiple local interfaces that are connected to
 different access networks.
 
+In case of conflicts between protocol and path selection properties,
+path selection properties take precedence.
+For example, if an application indicates a preference for a specific path, but
+also a preference for a protocol not available on this path, the transport
+system will try the path first, so the protocol selection property might not
+have an effect.
+
```
Comment (britram) [inline draft-trammell-taps-interface.md]: Precedence order (as above) should IMO appear here too...

File: draft-brunstrom-taps-impl.md (hunk 1 above)
Comment (theri) [reply draft-brunstrom-taps-impl.md]: Makes sense. I added more text to the implementation that details how this should work.

File: draft-trammell-taps-interface.md (hunk 2 above)
Comment (theri) [reply draft-trammell-taps-interface.md]: Agreed, I added some text here too, and moved it below the table. This section should perhaps be restructured in the future.

**Pull request merged**